packages/hooks/src/useAsyncEffect/index.ts

```diff
@@ -14,10 +14,10 @@ function useAsyncEffect(
     const e = effect();
     let cancelled = false;
     async function execute() {
-      if (isGenerator(e)) {
+      if (isAsyncGenerator(e)) {
         while (true) {
           const result = await e.next();
-          if (cancelled || result.done) {
+          if (result.done || cancelled) {
```

Comment:
In general, cancel is always false, and it will always come to the second judgment, so I think it is more appropriate to put this in the first judgment